src/Microsoft.VisualStudio.Services.Agent/JobServer.cs

```diff
@@ -142,12 +153,46 @@ public async Task<BlobIdentifierWithBlocks> UploadLogToBlobstorageService(Stream
             // Since we read this while calculating the hash, the position needs to be reset before we send this 
             blob.Position = 0;
 
-            using(var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
+            using (var blobClient = CreateArtifactsClient(_connection, default(CancellationToken)))
             {
                 return await blobClient.UploadBlocksForBlobAsync(blobId, blob, default(CancellationToken));
             }
         }
 
+        public async Task<(DedupIdentifier dedupId, ulong length)> UploadToBlobStore(bool verbose, string itemPath, CancellationToken cancellationToken)
+        {
+            var (dedupClient, clientTelemetry) = await DedupManifestArtifactClientFactory.Instance
+                    .CreateDedupClientAsync(verbose, (str) => Trace.Info(str), this._connection, cancellationToken);
+
+            // Create chunks and identifier
+            var chunk = await ChunkerHelper.CreateFromFileAsync(FileSystem.Instance, itemPath, cancellationToken, false);
+            var rootNode = new DedupNode(new []{ chunk});
+            var dedupId = rootNode.GetDedupIdentifier(HashType.Dedup64K);
```

Comment:
Where does `HashType.Dedup64k` come from? Is it worth calling that out in a comment?

Reply:
Done